src/ResourceManagement/CosmosDB/CosmosDBAccountImpl.cs

```diff
@@ -364,19 +374,16 @@ public CosmosDBAccountImpl WithSessionConsistency()
             return this;
         }
 
-        private void AddLocationsForParameters(HasLocation locationParameters, IList<Microsoft.Azure.Management.CosmosDB.Fluent.Models.FailoverPolicy> failoverPolicies)
+        private void AddLocationsForParameters(HasLocation locationParameters, IList<Microsoft.Azure.Management.CosmosDB.Fluent.Models.Location> failoverPolicies)
         {
             List<Models.Location> locations = new List<Models.Location>();
             if (failoverPolicies.Count > 0)
             {
-                for (int i = 0; i < failoverPolicies.Count; i++)
+                for (int i = 0; i < failoverPolicies.Count; ++i)
                 {
-                    Models.FailoverPolicy policyInner = failoverPolicies[i];
-                    Models.Location location = new Models.Location();
-                    location.FailoverPriority = i;
-                    location.LocationName = policyInner.LocationName;
-                    locations.Add(location);
+                    failoverPolicies[i].FailoverPriority = i;
```

Comment:
Just ask, priority=0 is highest or lowest?

Reply:
The 0 seems highest. Priority always from 0 ~ n - 1 in order from response.